### PR TITLE
[MSHARED-1150] Add ps1 as detectable file type for PowerShell

### PR DESCRIPTION
**Pull request by iofit**
To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [X ] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

### CODE REVIEW

File: src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java [hunk 1]
```diff
@@ -251,7 +251,7 @@ public void testShouldFindDummyMavenExecutable()
         File check;
         if ( Os.isFamily( Os.FAMILY_WINDOWS ) )
         {
-            check = createDummyFile( dummyMavenHomeBin, "mvn.bat" );
+            check = createDummyFile( dummyMavenHomeBin, "mvn.ps1" );
```
Comment (slawekjaranowski) [inline src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java:254]: `.bat` extensions still exists in code, so I would like a new test for `.ps1` extension

Comment (michael-o) [reply src/test/java/org/apache/maven/shared/invoker/MavenCommandLineBuilderTest.java:254]: Make sense

File: src/main/java/org/apache/maven/shared/invoker/MavenCommandLineBuilder.java [hunk 2]
```diff
@@ -663,6 +663,12 @@ private File detectMavenExecutablePerOs( File baseDirectory, String executable )
             {
                 return executableFile;
             }
+
+            executableFile = new File( baseDirectory, executable + ".ps1" );
+            if ( executableFile.isFile() )
+            {
+                return executableFile;
+            }
```
[the remark this answers is not in the document]
Comment (iofit) [reply src/main/java/org/apache/maven/shared/invoker/MavenCommandLineBuilder.java]: I moved up ps1 then check for cmd and bat

**Review by slawekjaranowski — APPROVED** [review]

**Review by michael-o — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #139

**Pull request merged**